app/src/components/modals/AddCompetitionModal.tsx

```diff
@@ -0,0 +1,334 @@
+import { Alert, Platform, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native";
+import { StandardButton } from "../../ui/StandardButton.tsx";
+import { useTheme } from "@react-navigation/native";
+import { useEffect, useState } from "react";
+import RNDateTimePicker from "@react-native-community/datetimepicker";
+import { StandardModal } from "./StandardModal.tsx";
+import { supabase } from "../../lib/supabase.ts";
+import { FormSelect } from "../../ui/form/components/FormSelect.tsx";
+import { getIdealTextColor, parseColor } from "../../lib/color.ts";
+import { FunctionsHttpError } from "@supabase/supabase-js";
+import type { Setter } from "../../lib/react";
+
+function Spacer() {
+    return <View style={{ height: "2%" }} />;
+}
+
+export interface AddCompetitionModalProps {
+    visible: boolean;
+    setVisible: Setter<boolean>;
+    onRefresh: () => void;
+}
+export function AddCompetitionModal({ visible, setVisible, onRefresh }: AddCompetitionModalProps) {
```

Comment:
i mean if we're going to do es6 functions everywhere else then why not just be consistant ykwim